pyathena/sqlalchemy/base.py

```diff
@@ -435,6 +435,29 @@ def limit_clause(self, select: "GenerativeSelect", **kw):
             text.append(" LIMIT " + self.process(select._limit_clause, **kw))
         return "\n".join(text)
 
+    def get_from_hint_text(self, table, text):
+        """Gets the text of a FROM clause hint"""
+        return text
+
+    def format_from_hint_text(self, sqltext, table, hint, iscrud):
+        """ Parses the text of a FROM clause hint, returning the text formatted properly"""
+        hint_upper = hint.upper()
+
+        if not any([
+            hint_upper.startswith("FOR TIMESTAMP AS OF"),
+            hint_upper.startswith("FOR SYSTEM_TIME AS OF"),
+            hint_upper.startswith("FOR VERSION AS OF"),
+            hint_upper.startswith("FOR SYSTEM_VERSION AS OF"),
+        ]):
+            raise exc.CompileError("Unrecognized hint: %r" % hint)
```

Comment:
There are other possible uses of the hint clause, so do not raise the error.

Reply:
Oh, I thought these were just for the selectable from clause. I had to add the `get_from_hint_text` to get them to go past the compilation step. They would show up in the string, but once the PyAthena dialect got involved it didn't leverage them in the compile.  Do you want to remove this check entirely or add the other supported hints?

Reply:
This change is not the original use of the hint clause. I cannot accept any change that would cause an error except for the time travel setting.
Athena will return an error if there is an error in the SQL syntax, so there is no need to check this in detail.

Reply:
Updated based on this feedback.